crates/wasm/src/lib.rs

```diff
@@ -28,21 +28,45 @@ pub fn compile(src: String) -> JsValue {
     let compiled_program = noirc_driver::Driver::compile_file(path, language);
     <JsValue as JsValueSerdeExt>::from_serde(&compiled_program).unwrap()
 }
+
 // Deserializes bytes into ACIR structure
+#[deprecated(
+    note = "we have moved away from this serialization strategy. Call `acir_read_bytes` instead"
```

Comment:
Why? What is the difference?

Reply:
It's using our new handwritten serialisation rather than rmpserde + compression.

Reply:
The `to_bytes` and `from_bytes` methods are already marked as deprecated in the acvm: https://github.com/noir-lang/acvm/blob/373c18fc05edf673cfec9e8bbb78bd7d7514999e/acir/src/circuit/mod.rs#L35.

And the most up-to-date nargo has move to use our native serialization methods: https://github.com/noir-lang/noir/blob/6c7aa2fc39c7caff1fee94888287f17101101e43/crates/nargo/src/cli/compile_cmd.rs#L94